msal/application.py

```diff
@@ -973,6 +974,11 @@ def authorize():  # A controller in a web app
             **kwargs))
         if "access_token" in response:
             response[self._TOKEN_SOURCE] = self._TOKEN_SOURCE_IDP
+        if ("AADSTS500207" in response.get("error_description", "") and
+                self._oidc_authority and not self._oidc_authority.endswith("/v2.0")):
+            response["error_description"] = (
+                'Did you forget to append "/v2.0" to your oidc_authority? '
```

Comment:
I feel like the best way that this error should be handled/prevented is for the v2.0 endpoint not even existing for CIAM; which is being worked on by the CIAM team. You could choose to keep this handling here, but just know that in the near future developers should never be getting this error anymore as being caused by forgetting to add v2.0 to the URL path.

Reply:
@SammyO - does the CIAM team consider making this change for CIAM CUD GA? Because once we go GA, it's all over - you'll have to support "v2.0" suffix (maybe through DNS hackery), otherwise you'll break production apps.

Reply:
@bgavrilMS yes, exactly because of the reason you've outlined there.

Reply:
@rayluo - pls open a similar issue to MSAL.NET

Reply:
> the v2.0 endpoint not even existing for CIAM; which is being worked on by the CIAM team

Does that mean the OIDC authority for CUD after its GA will be simply `https://contoso.com/tenant` i.e. without the `/v2.0` suffix? That would essentially mean the V2 URL converges into the V1 URL. We had this proposal before, but it was rejected because it is believed that V1 and V2 URLs need to coexist. Is that no longer the case? @SammyO

Reply:
@rayluo what I think will be happening is a v1.0 version of an authority endpoint returning something like a 404. I don't think there will be a converging (as this would hide logic from developers and make things less explicit, which IMO is not desirable). But let me confirm.